enterprise/views.py

```diff
@@ -1137,3 +1119,91 @@ def post(self, request, enterprise_uuid, program_uuid):
             )
 
         return redirect(basket_page)
+
+
+class RouterView(NonAtomicView):
+    """
+    A router or gateway view for managing Enterprise workflows.
+    """
+
+    # pylint: disable=invalid-name
+    COURSE_ENROLLMENT_VIEW_URL = '/enterprise/{}/course/{}/enroll/'
+    PROGRAM_ENROLLMENT_VIEW_URL = '/enterprise/{}/program/{}/enroll/'
+    HANDLE_CONSENT_ENROLLMENT_VIEW_URL = '/enterprise/handle_consent_enrollment/{}/course/{}/'
+    VIEWS = {
+        COURSE_ENROLLMENT_VIEW_URL: CourseEnrollmentView,
+        PROGRAM_ENROLLMENT_VIEW_URL: ProgramEnrollmentView,
+        HANDLE_CONSENT_ENROLLMENT_VIEW_URL: HandleConsentEnrollment,
+    }
+
+    @staticmethod
+    def get_path_variables(**kwargs):
+        """
+        Get the base variables for any view to route to.
+
+        Currently gets:
+        - `enterprise_uuid` - the UUID of the enterprise customer.
+        - `course_run_id` - the ID of the course, if applicable.
+        - `program_uuid` - the UUID of the program, if applicable.
+        """
+        enterprise_customer_uuid = kwargs.get('enterprise_uuid', '')
+        course_run_id = kwargs.get('course_id', '')
+        program_uuid = kwargs.get('program_uuid', '')
+        return enterprise_customer_uuid, course_run_id or program_uuid
+
+    def eligible_for_direct_audit_enrollment(self, request, enterprise_customer, course_run_id):
```

Comment:
Can we add a check to make sure that `audit` is a valid enrollment mode for the given course?

Reply:
Yes :+1:

Reply:
Addressed with c9d516a4f7b0ce80bacfb93cf6088648973887f2

Reply:
Forgot to update docstring, though.